File: Easy/ReverseLinkedList.cpp

struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution
{
public:
    ListNode *reverseList(ListNode *head)
    {
        if (head == nullptr)
            return nullptr;

        ListNode *temp = head;
        ListNode *newHead = new ListNode(head->val);

        while (temp->next != nullptr)
        {
            ListNode *node = new ListNode(temp->next->val, newHead);
            newHead = node;
            temp = temp->next;
        }

        return newHead;
    }
};

File: Python/Easy/ReverseLinkedList.py
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None:
            return None
        
        temp = head

        newHead = ListNode(head.val)

        while temp.next is not None:
            node = ListNode(temp.next.val, newHead)
            newHead = node
            temp = temp.next

        return newHead